ait/core/server/server.py

```diff
@@ -0,0 +1,255 @@
+            return PortOutputStream(name,
+                                    stream_input,
+                                    stream_output,
+                                    stream_handlers,
+                                    zmq_args={'zmq_context': self.broker.context,
+                                              'zmq_proxy_xsub_url': self.broker.XSUB_URL,
+                                              'zmq_proxy_xpub_url': self.broker.XPUB_URL})
+        elif type(stream_input) is int:
+            return PortInputStream(name,
+                                   stream_input,
+                                   stream_handlers,
+                                   zmq_args={'zmq_context': self.broker.context,
+                                             'zmq_proxy_xsub_url': self.broker.XSUB_URL,
+                                             'zmq_proxy_xpub_url': self.broker.XPUB_URL})
+        else:
+            return ZMQInputStream(name,
+                                  stream_input,
+                                  stream_handlers,
+                                  zmq_args={'zmq_context': self.broker.context,
+                                            'zmq_proxy_xsub_url': self.broker.XSUB_URL,
+                                            'zmq_proxy_xpub_url': self.broker.XPUB_URL})
+
+    def _create_handler(self, config):
+        """
+        Creates a handler from its config.
+
+        Params:
+            config:      handler config
+        Returns:
+            handler instance
+        """
+        if config is None:
+            raise ValueError('No handler config to create handler from.')
+
+        if 'name' not in config:
+            raise ValueError('Handler name is required.')
+
+        handler_name = config['name']
+        # try to create handler
+        module_name = handler_name.rsplit('.', 1)[0]
+        class_name = handler_name.rsplit('.', 1)[-1]
+        module = import_module(module_name)
+        handler_class = getattr(module, class_name)
+        instance = handler_class(**config)
+
+        return instance
+
+    def _load_plugins(self):
+        """
+        Reads, parses and creates plugins specified in config.yaml.
+        """
+        plugins = ait.config.get('server.plugins')
+
+        if plugins is None:
+            log.warn('No plugins specified in config.')
+        else:
+            for index, p in enumerate(plugins):
+                try:
+                    plugin = self._create_plugin(p['plugin'])
+                    self.plugins.append(plugin)
+                    log.info('Added plugin {}'.format(plugin))
+
+                except Exception:
+                    exc_type, value, tb = sys.exc_info()
+                    log.error('{} creating plugin {}: {}'.format(exc_type,
+                                                                 index,
+                                                                 value))
+            if not self.plugins:
```

Comment:
This check seems a bit off to me. Seems like a double check with the previous if/else block that isn't necessary. Maybe something that got left in and needs cleaned out?

Reply:
This is for the case where plugins are specified in the config, but none of those configs are valid